Relative filters in configuration files are resolved incorrectly

**Describe the bug**

I'm trying to use `gcovr.cfg` to load dynamic exclude rules, but the relative file parameter is not working as described in the docu:
```
Load that configuration file. Defaults to gcovr.cfg in the --root directory.
```
If i do not add any parameter the file `gcovr.cfg` in root is not loaded, because excludes are not applied.
Therefore i've added the option `--config gcovr.cfg` and face error message, see below.

**To Reproduce**
Steps to reproduce the behavior:
1. no error, but `gcovr.cfg` is not loaded : `/usr/local/bin/gcovr --html --html-details -r /home/andre/git/project --object-directory=/home/andre/git/project/build/coverage -o test.coverage.html/index.html`

1. add one parameter `--config gcovr.cfg` : `/usr/local/bin/gcovr --html --html-details -r /home/andre/git/project --config gcovr.cfg --object-directory=/home/andre/git/project/build/coverage -o test.coverage.html/index.html`

now the invocation returns error:
```
Traceback (most recent call last):
  File "/usr/local/bin/gcovr", line 8, in <module>
    sys.exit(main())
  File "/home/andre/.local/lib/python3.9/site-packages/gcovr/__main__.py", line 275, in main
    collect_coverage_from_gcov(covdata, options, logger)
  File "/home/andre/.local/lib/python3.9/site-packages/gcovr/__main__.py", line 326, in collect_coverage_from_gcov
    datafiles.update(find_files(search_path, logger, options.exclude_dirs))
  File "/home/andre/.local/lib/python3.9/site-packages/gcovr/gcov.py", line 64, in find_datafiles
    files = list(search_file(
  File "/home/andre/.local/lib/python3.9/site-packages/gcovr/utils.py", line 48, in search_file
    raise IOError("Unknown directory '" + path + "'")
OSError: Unknown directory '--config gcovr.cfg'
ninja: build stopped: subcommand failed.
```

**Expected behavior**
The file `gcovr.cfg` should loaded relative to root as described in the docu.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Desktop (please complete the following information):**
 - OS: Linux Debian 11
 - GCC version: GCC 11.1
 - GCOVR version: 5.0
 - Project directory layout:
   - Roots: /home/andre/git/project
   - Object directories: /home/andre/git/project/build/coverage

**Additional context**

- second issue is with understanding how excludes are handled - if it would work relative to location of the config file then this should exclude the two folders:
```
exclude = build/*
exclude = libs/*
```
but it does not.

The exclude is working at command line via absolute paths `-e /home/andre/git/plcapp/build/*` and `-e /home/andre/git/plcapp/libs/*` :
```
/usr/local/bin/gcovr --html --html-details -r /home/andre/git/plcapp -e /home/andre/git/plcapp/build/* -e /home/andre/git/plcapp/libs/* --object-directory=/home/andre/git/plcapp/build/coverage -o test.coverage.html/index.html
```

so how to achive this via `gcovr.cfg` ?

- `exclude = build/*` same like `-e /home/andre/git/plcapp/build/*`
- `exclude = libs/*` same like `-e /home/andre/git/plcapp/libs/*`

i've tried to use variable `${CMAKE_SOURCE_DIR}` set by cmake:
```
exclude = ${CMAKE_SOURCE_DIR}/build/*
exclude = ${CMAKE_SOURCE_DIR}/libs/*
```

but it shows:
```
Traceback (most recent call last):
  File "/usr/local/bin/gcovr", line 8, in <module>
    sys.exit(main())
  File "/home/andre/.local/lib/python3.9/site-packages/gcovr/__main__.py", line 139, in main
    cfg_options = parse_config_into_dict(
  File "/home/andre/.local/lib/python3.9/site-packages/gcovr/configuration.py", line 378, in parse_config_into_dict
    for cfg_entry in config_entry_source:  # type: ConfigEntry
  File "/home/andre/.local/lib/python3.9/site-packages/gcovr/configuration.py", line 1064, in parse_config_file
    raise error("variable substitution syntax "
SyntaxError: /home/andre/git/plcapp/gcovr.cfg: 2: variable substitution syntax (${var}, $(var), or $var) is reserved
on this line: exclude = ${CMAKE_SOURCE_DIR}/build/*
ninja: build stopped: subcommand failed.
```

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -24,6 +24,7 @@ Improvements and new features:
 - Generate also a Report of Excluded Coverage. (:issue:`503`)
 - Added :option:`--decisions` to add decision coverage to HTML and JSON output. (:issue:`350`)
 - Replace own logger with python logger module. (:issue:`540`)
+- Fix resolution of relative filters in configuration files. (:issue:`568`)
 
 Documentation:
 

--- a/gcovr/configuration.py
+++ b/gcovr/configuration.py
@@ -17,6 +17,7 @@
 # ****************************************************************************
 
 from argparse import ArgumentTypeError, SUPPRESS
+from inspect import isclass
 from locale import getpreferredencoding
 from multiprocessing import cpu_count
 from typing import Iterable, Any
@@ -429,7 +430,7 @@ def _get_value_from_config_entry(cfg_entry, option):
 
         value = cfg_entry.value
         args = ()
-        if option.type is FilterOption:
+        if isclass(option.type) and issubclass(option.type, FilterOption):
             args = [os.path.dirname(cfg_entry.filename)]
         elif option.type is check_input_file:
             value = check_input_file(value, os.path.dirname(cfg_entry.filename))

